internal/node/node_launcher.sh

```diff
@@ -129,4 +129,22 @@ for ARG in "${ALL_ARGS[@]}"; do
   esac
 done
 
-exec "${node}" "${NODE_OPTIONS[@]}" "${script}" "${ARGS[@]}"
+set +e
+"${node}" "${NODE_OPTIONS[@]}" "${script}" "${ARGS[@]}"
+RESULT="$?"
+set -e
+
+readonly EXPECTED_EXIT_CODE="TEMPLATED_expected_exit_code"
+if [ "${EXPECTED_EXIT_CODE}" -ne "0" ]; then
+  if (( ${RESULT} != ${EXPECTED_EXIT_CODE} )); then
+    echo "Expected exit code to be ${EXPECTED_EXIT_CODE}, but got ${RESULT}" >&2
+    # This exit code is handled specially by Bazel:
+    # https://github.com/bazelbuild/bazel/blob/486206012a664ecb20bdb196a681efc9a9825049/src/main/java/com/google/devtools/build/lib/util/ExitCode.java#L44
+    readonly BAZEL_EXIT_TESTS_FAILED = 3;
```

Comment:
I'm a little surprised that this talks about test handling - what if I just `bazel run :my_binary` and look for the exit code 128 as listed in the documentation for my_binary?
I think we either need to:
1) check here for whether we are in a test, like whether $TEST_SRCDIR is defined
2) just exit with the same code as the program - it's up to the program we are running to be aware of Bazel exit codes if it's a test.

I think I prefer #2

Reply:
oh I see, if actual exit code is zero then using BAZEL_EXIT_TESTS_FAILED seems reasonable.